src/mappings/handlers/multisigEventHandler.ts

```diff
@@ -0,0 +1,157 @@
+import { SubstrateEvent } from "@subql/types";
+import { u8aToHex } from "@polkadot/util";
+import { decodeAddress } from "../../utils";
+import { CreateCallVisitorBuilder, CreateCallWalk, VisitedCall } from "subquery-call-visitor";
+import { EventStatus, MultisigEvent, MultisigOperation, OperationStatus } from "../../types";
+import {
+  generateEventId,
+  generateOperationId,
+  getBlockCreated,
+  getDataFromCall,
+  getDataFromEvent,
+  getIndexCreated,
+  timestamp,
+} from "../../utils/operations";
+import { AccountId, DispatchResult, Timepoint } from "@polkadot/types/interfaces";
+import { AnyTuple, CallBase } from "@polkadot/types/types";
+
+const callWalk = CreateCallWalk();
+
+export async function handleMultisigCallEvent(event: SubstrateEvent) {
+  if (!event.extrinsic) throw new Error("Extrinsic not found");
+
+  const isNewMultisig = event.event.method === "NewMultisig";
+
+  const possibleIndex = isNewMultisig ? 2 : 3;
+  const callHash = getDataFromEvent<Uint8Array>(event.event, "callHash", possibleIndex);
+
+  if (!callHash) throw new Error("Call hash not found");
+
+  const callHashString = u8aToHex(callHash);
+
+  const multisigPossibleIndex = isNewMultisig ? 1 : 2;
+  const multisig = getDataFromEvent<AccountId>(event.event, "multisig", multisigPossibleIndex);
+
+  if (!multisig) throw new Error("Multisig not found");
+
+  const multisigAccountId = multisig.toHex();
+
+  let timepoint = null;
+  let blockCreated = getBlockCreated(event.extrinsic);
+  let indexCreated = getIndexCreated(event.extrinsic);
+  if (!isNewMultisig) {
+    timepoint = getDataFromEvent<Timepoint>(event.event, "timepoint", 1) || getDataFromEvent<Timepoint>(event.event, "Timepoint", 1);
+  }
```

Comment:
I think we should otherwise set a timpoint based on block number and extrinsic index. Otherwise it will be null untill second approval